### PR TITLE
Upgrade to discord.py 1.1.0

### LINKED ISSUE #186: Address commands.Cog breaking changes
The rewrite branch recently got a new update with the way how cogs work rewritten.

Changes: https://discordapp.com/channels/336642139381301249/381965829857738772/548794992395550733

# Checklist 
- [ ] Update Internal Code
- [ ] Update Plugin Examples 
- [ ] Update requirements.txt and push version

### PR DESCRIPTION
**Pull request by kyb3r**
We should use the latest version of discord.py and also update related documentation for plugins.

Resolves #186

### CODE REVIEW

**Comment by fourjr** [comment]
Lock version and use `discord.py` not the github version.

**Comment by fourjr** [comment]
@Taaku18 Does this still work? If so, we need to merge this next.

v1.1.0 has been recently pushed to PyPi.

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-git+https://github.com/Rapptz/discord.py@7f4c57dd5ad20b7fa10aea485f674a4bc24b9547
+git+https://github.com/Rapptz/discord.py
```
Comment (fourjr) [inline requirements.txt]: lock to `discord.py===xx`

**Comment by Taaku18** [comment]
> @Taaku18 Does this still work? If so, we need to merge this next.
> 
> v1.1.0 has been recently pushed to PyPi.

I don't believe so, it's time to look into this now.

**Comment by Taaku18** [comment]
There're still some problems, for example, command signatures were changed in v1.0 rendering some help commands broken. A possible remedy is to look into the new Help Command framework provided by discord.py https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#help-commands.

**Comment by fourjr** [comment]
Is this ready for merge into master?

**Comment by Taaku18** [comment]
> Is this ready for merge into master?

Almost, give me one more day, gonna test a bit more and I accidentally removed uvloop.

**Comment by Taaku18** [comment]
I believe everything's fine now, have a look through, and if everything's good, merge is ready.

**Comment by officialpiyush** [comment]
:tada:

**Pull request merged**